doc: add commands (#21)
re-vault/revaultd: doc/API.md

revaultd exposes a [JSON-RPC 2.0](https://www.jsonrpc.org/specification)
interface over a Unix Domain socket.

| Command                                   | Description                                          |
| ----------------------------------------- | ---------------------------------------------------- |
| [`getinfo`](#getinfo)                     | Display general information                          |
| [`listvaults`](#listvaults)               | Display a paginated list of vaults                   |
| [`getrevocationtxs`](#getrevocationtxs)   | Retrieve the Revault revocation transactions to sign |
| [`getunvaulttx`](#getunvaulttx)           | Retrieve the Revault unvault transaction to sign     |
| [`getspendtx`](#getspendtx)               | Retrieve the Revault spend transaction to sign       |
| [`revocationtxs`](#revocationtxs)         | Give back the revocation transactions signed         |
| [`unvaulttx`](#unvaulttx)                 | Give back the unvault transaction signed             |

> | [`spendtx`](#spendtx)                     | Give back the spend transaction signed               |

Reviewer: It looks like a *nice* interface to me :+1:

| ------------- | ------------------------------------------ | ------------------------- |
| `vaults`      | array of [vault resource](#vault-resource) | Vaults filtered by status |

> ### `securevault`

Author: I will change it for `get_revocation_txs` as suggested below

The `securevault` RPC Command executes the signing process of the
Revault revocation transactions.

#### Request

| Parameter        | Type    | Description                                     |
| ---------------- | ------- | ----------------------------------------------- |
| `txid`           | string  | Unique ID of the vault to secure                |

> | `only_emergency` | boolean | Retrieve only the emergency transaction to sign |

Author: OK to remove it

#### Response

| Field                | Type   | Description                                                 |
| -------------------- | ------ | ----------------------------------------------------------- |
| emergency_tx         | string | Emergency transaction to sign using the PSBT format         |
| cancel_tx            | string | Cancel transaction to sign using the PSBT format            |
| emergency_unvault_tx | string | Emergency unvault transaction to sign using the PSBT format |

### `activatevault`

The `activatevault` RPC Command executes the signing process of the
Revault unvault transaction.

#### Request

| Parameter        | Type    | Description                        |
| ---------------- | ------- | ---------------------------------- |

> | `txid`           | string  | Unique ID of the vault to activate |

Reviewer: You missed this one, but this can be added later

| ------------- | ------------------------------------------ | ------------------------- |
| `vaults`      | array of [vault resource](#vault-resource) | Vaults filtered by status |

### `getrevocationtxs`

> The `getrevocationtxs` RPC Command executes the signing process of the
> Revault revocation transactions.

Reviewer: ```suggestion
The `getrevocationtxs` RPC Command builds and return the revocation
transactions corresponding to this vault.
```

#### Request

| Parameter        | Type    | Description                                     |
| ---------------- | ------- | ----------------------------------------------- |

> | `txid`           | string  | Unique ID of the vault to secure                |

Reviewer: nit: "Deposit txid"

#### Response

| Field                  | Type   | Description                                                 |
| ---------------------- | ------ | ----------------------------------------------------------- |
| `emergency_tx`         | string | Emergency transaction to sign using the PSBT format         |
| `cancel_tx`            | string | Cancel transaction to sign using the PSBT format            |
| `emergency_unvault_tx` | string | Emergency unvault transaction to sign using the PSBT format |

### `revocationtxs`

> The PSBTs once signed must be given back to the daemons.

Reviewer: Nit: "daemon" (withtout the `s`)

#### Request

| Field                  | Type   | Description                                                |
| ---------------------- | ------ | ---------------------------------------------------------- |
| `emergency_tx`         | string | Emergency transaction signed using the PSBT format         |
| `cancel_tx`            | string | Cancel transaction signed using the PSBT format            |
| `emergency_unvault_tx` | string | Emergency unvault transaction signed using the PSBT format |

### `getunvaulttx`

> The `getunvaulttx` RPC Command executes the signing process of the
> Revault unvault transaction.

Reviewer: ```suggestion
The `getunvaulttx` RPC Command builds and return the unvault
transaction corresponding to this vault.
```

### `signedtx`

The PSBT once signed must be given back to the daemon to be handled
according to the context. 

If the `signedtx` type is `spend_tx` and has enough manager signatures, 
then the revault daemon will start the spending process. 

#### Request

| Parameter | Type    | Description                                                                                                   |
| --------- | ------- | ------------------------------------------------------------------------------------------------------------- |
| tx        | string  | PSBT with the required signatures                                                                             |
| type      | string  | Type of the transaction, can be `emergency_tx`, `cancel_tx`, `emergency_unvault_tx`, `unvault_tx`, `spend_tx` |

#### Response

| Field    | Type | Description                                                                                             |
| -------- | ---- | ------------------------------------------------------------------------------------------------------- |
| sigs_ack | bool | Watchower has stored the given revocation tx (Field not present if tx type is `unvault_tx`, `spend_tx`) |

> 

Author: Ok I agree that the call would be more explicit with `spend_tx` etc

```

## Manager flow

```
HSM                client                      revaultd
  +                      +                          +
  |                      | +---listvaults active--> |
  |                      | <-------vaults---------+ |
  |                      |                          |
  |                      | +---spendvaults--------> |
  |                      | <-----psbt or wt nack--+ |
  | <----sign spend tx-+ |                          |
  | +------sig---------> |                          |
  +                      |                          |
                         |                          |
client 2                 |                          |
  +                      |                          |
  | <---sign psbt------+ |                          |
  | +-----psbt---------> |                          | // daemon ask wt opinion
  +                      | +-------signtx psbt----> | // if ack, cosign server sign

>                          | <---OK or wt nack------+ | // then daemon broadcasts tx

Reviewer: Note that WT opinion has already been polled, and final validation is done *after* requesting the sig from the cosig
